core/src/test/kotlin/io/specmatic/core/pattern/AnyOfPatternTest.kt

```diff
@@ -0,0 +1,342 @@
+    }
+
+    @Test
+    fun `should encompass JSONObjectPattern with compatible structure`() {
+        val objectPattern1 = JSONObjectPattern(mapOf("name" to StringPattern(), "age" to NumberPattern()), typeAlias = "(Person)")
+        val objectPattern2 = JSONObjectPattern(mapOf("id" to NumberPattern()), typeAlias = "(Id)")
+        val anyOfPattern = AnyOfPattern(listOf(objectPattern1, objectPattern2), extensions = emptyMap())
+        
+        val compatiblePattern = JSONObjectPattern(mapOf("name" to StringPattern()), typeAlias = "(SimplePerson)")
+        
+        val result = anyOfPattern.encompasses(compatiblePattern, Resolver(), Resolver())
+        assertThat(result).isInstanceOf(Result.Success::class.java)
+    }
+
+    @Test
+    fun `should fix value using best matching pattern`() {
+        val pattern = AnyOfPattern(listOf(NumberPattern(), StringPattern()), extensions = emptyMap())
+        val value = StringValue("hello")
+        
+        val fixedValue = pattern.fixValue(value, Resolver())
+        assertThat(fixedValue).isEqualTo(value)
+    }
+
+    @Test
+    fun `should eliminate optional keys correctly`() {
+        val objectPattern = parsedPattern("""{
+            "mandatoryKey": "(string)",
+            "optionalKey?": "(number)"
+        }""")
+        val pattern = AnyOfPattern(listOf(objectPattern), extensions = emptyMap())
+        
+        val originalValue = parsedValue("""{
+            "mandatoryKey": "hello",
+            "optionalKey": 42
+        }""")
+        
+        val expectedValue = parsedValue("""{
+            "mandatoryKey": "hello"
+        }""")
+        
+        val result = pattern.eliminateOptionalKey(originalValue, Resolver())
+        assertEquals(expectedValue, result)
+    }
+
+    @Test
+    @Tag(GENERATION)
+    fun `should generate new patterns for all available types`() {
+        AnyOfPattern(
+            listOf(
+                NumberPattern(),
+                EnumPattern(listOf(StringValue("one"), StringValue("two")))
+            ),
+            extensions = emptyMap()
+        ).newBasedOn(Row(), Resolver()).map { it.value }.toList().let { patterns ->
+            patterns.map { it.typeName } shouldContainInAnyOrder listOf("number", "\"one\"", "\"two\"")
+        }
+    }
+
+    @Test
+    @Tag(GENERATION) 
+    fun `should create new pattern based on row data`() {
+        val pattern = AnyOfPattern(listOf(NumberPattern(), StringPattern()), extensions = emptyMap())
+        val row = Row(listOf("hello"))
+        
+        val newPatterns = pattern.newBasedOn(row, Resolver()).toList()
+        assertThat(newPatterns).hasSize(2)
+        
+        val numberPattern = newPatterns.find { it.value?.typeName == "number" }
+        val stringPattern = newPatterns.find { it.value?.typeName == "\"hello\"" }
+        
+        assertThat(numberPattern).isNotNull
+        assertThat(stringPattern).isNotNull
+    }
+
```

Comment:
create a newBasedOn test for anyOf where
- the anyOf has two JSONObjectPattern objects
- the row as an example matching one of the patterns